recipes/leveldb/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LevelDBConan(ConanFile):
+    name = "leveldb"
+    description = ("LevelDB is a fast key-value storage library written at "
+                   "Google that provides an ordered mapping from string keys "
+                   "to string values")
+    license = "https://github.com/google/leveldb/blob/master/LICENSE"
+    topics = ("conan", "leveldb", "google", "db")
+    url = "https://github.com/conan-io/conan-center-index "
+    homepage = "https://github.com/google/leveldb"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_snappy": [True, False],
+    }
+    default_options = {
+            "shared": False,
+            "fPIC": False,
+            "with_snappy": False,
+    }
+    generators = "cmake"
+
+    _cmake = None
+
+    def _get_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LEVELDB_BUILD_TESTS"] = False
+        if self.options.fPIC:
+            self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = True
+        return self._cmake
+
+    # note: crc32, tcmalloc are also conditionally included in leveldb, but
+    # there are no "official" conan packages yet; when those are available, we
+    # can add similar with- options for those
+    optional_snappy_requirement = "snappy/1.1.7"
+
+    def requirements(self):
+        if self.options.with_snappy:
+            self.requires(self.optional_snappy_requirement)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        downloaded_name = "%s-%s" % (self.name, self.version)
+        os.rename(downloaded_name, self._source_subfolder)
+        with tools.chdir(self._source_subfolder):
+            if not self.options.with_snappy:
+                tools.replace_in_file(
+                    "CMakeLists.txt",
+                    ('''check_library_exists(snappy snappy_compress '''
+                     '''"" HAVE_SNAPPY)'''),
+                    ('''check_library_exists(snappy snappy_compress '''
+                        '''"" IGNORE_HAVE_SNAPPY)'''))
+
+    def build(self):
+        cmake = self._get_cmake()
+        cmake.configure(source_folder=self._source_subfolder)
+        cmake.build()
+
+    def package(self):
+        cmake = self._get_cmake()
+        self.copy("LICENSE", src=self._source_subfolder,
+                  dst="licenses", keep_path=False)
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
Does it provide only one library?
Interdependencies between the libraries can break the build so it's often better to explicitly list the libraries.
Even if it's only one library.
Such as
```python
self.cpp_info.libs = ["leveldb"]
```

Reply:
Yes just one library. Will do.